vtex/faststore
# Create 'Bullets' molecule

## What's the purpose of this pull request?


This PR adds the **Bullets** molecule to our store-ui library.

## How it works? 


The `Bullets` molecule is supposed to be used in scenarios where we have paginated items, which can be navigated through the use of small bullets indicating where the user is and where they can go to. This is commonly used along with Carousels, such as this example:

![](https://blog.logrocket.com/wp-content/uploads/2019/08/building-carousel-react-using-only-hook-nocdn.jpg)

The `Bullets` molecule would be used to make those small circles at the bottom of the image. And it supports the following props:

```ts
interface BulletsProps {
  /**
   * Number of bullets that should be rendered.
   */
  totalQuantity: number
  /**
   * The currently active bullet (zero-indexed).
   */
  activeBullet: number
  /**
   * Event handler for clicks on each bullet. The handler will be called with
   * the index of the bullet that received the click.
   */
  onClick: (e: MouseEvent, bulletIdx: number) => void
  /**
   * ID to find this component in testing tools (e.g.: cypress,
   * testing-library, and jest).
   */
  testId?: string
  /**
   * Function that should be used to generate the aria-label attribute added
   * to each bullet that is inactive. It receives the bullet index as an
   * argument so that it can be interpolated in the generated string.
   */
  ariaLabelGenerator?: (index: number, isActive: boolean) => string
}
```

To enable different CSS rules to be applied to the current bullet set as 'active', we render the currently active bullet with a data attribute `data-active`.

## How to test it?


You can check the deploy preview linked below, where you'll find a `Bullets` section listed under `Molecules` in our Storybook.
Take a look at the `Story` section to see the code being used to control the `Bullets` molecule.
